src/app/ApplicationTemplate.Shared/Presentation/Authentication/CreateAccountFormViewModel.cs

```diff
@@ -20,19 +43,111 @@ public string Email
 			set => this.Set(value);
 		}
 
+		public string PhoneNumber
+		{
+			get => this.Get<string>();
+			set => this.Set(value);
+		}
+
+		public string SecondaryPhoneNumber
+		{
+			get => this.Get<string>();
+			set => this.Set(value);
+		}
+
+		public string PostalCode
+		{
+			get => this.Get<string>();
+			set => this.Set(value);
+		}
+
+		public DateTimeOffset DateOfBirth
+		{
+			get => this.Get<DateTimeOffset>(initialValue: DateTimeOffset.Now);
+			set => this.Set(value);
+		}
+
 		public string Password
 		{
 			get => this.Get<string>();
 			set => this.Set(value);
 		}
+
+		public string ConfirmPassword
+		{
+			get => this.Get<string>();
+			set => this.Set(value);
+		}
+
+		public string[] FavoriteDonuts
+		{
+			get => this.Get<string[]>();
+			set => this.Set(value);
+		}
+
+		public bool AgreeToTermsOfServices
+		{
+			get => this.Get<bool>();
+			set => this.Set(value);
+		}
 	}
 
 	public class CreateAccountFormValidator : AbstractValidator<CreateAccountFormViewModel>
 	{
-		public CreateAccountFormValidator()
+		public CreateAccountFormValidator(IStringLocalizer localizer)
 		{
+			var myPassword = string.Empty;
+
+			RuleFor(x => x.FirstName).NotEmpty();
+			RuleFor(x => x.LastName).NotEmpty();
 			RuleFor(x => x.Email).NotEmpty().EmailAddress();
-			RuleFor(x => x.Password).NotEmpty();
+			RuleFor(x => x.PhoneNumber).NotEmpty().Length(14).WithMessage(localizer["CreateAccount_PhoneNumberValidation"]); // "(000) 000-0000".Length = 14
+			RuleFor(x => x.SecondaryPhoneNumber).Must(phonenumber =>
+			{
+				if (phonenumber.IsNullOrEmpty())
+				{
+					return true;
+				}
+				else if (phonenumber.Length == 14)
+				{
+					return true;
+				}
+				else
+				{
+					return false;
+				}
+
+			}).WithMessage(localizer["CreateAccount_PhoneNumberValidation"]);
+			RuleFor(x => x.PostalCode).NotEmpty().Length(7).WithMessage(localizer["CreateAccount_PostalValidation"]); // "A1A 1A1".Length = 7
+			RuleFor(x => x.DateOfBirth).NotEmpty().Must(dateOfBirth =>
+			{
+				var diff = DateTime.Now.Year - dateOfBirth.DateTime.Year;
+				return diff > 18;
+			}).WithMessage(localizer["CreateAccount_DateOfBirthValidation"]);
+			RuleFor(x => x.Password).NotEmpty().Must(password =>
```

Comment:
When you have 2 or more extension chained, can you put each extension method on its own line, starting with the "."